tests/integration/telemetry/util.go

```diff
@@ -39,3 +43,24 @@ func PromDumpWithAttributes(cluster cluster.Cluster, prometheus prometheus.Insta
 
 	return ""
 }
+
+// Get trust domain of the cluster.
+func GetTrustDomain(cluster cluster.Cluster, istioNamespace string) string {
+	meshConfigMap, err := cluster.CoreV1().ConfigMaps(istioNamespace).Get(context.Background(), "istio", metav1.GetOptions{})
+	defaultTrustDomain := mesh.DefaultMeshConfig().TrustDomain
+	if err != nil {
+		return defaultTrustDomain
+	}
+
+	configYaml, ok := meshConfigMap.Data["mesh"]
+	if !ok {
+		return defaultTrustDomain
+	}
+
+	cfg, err := mesh.ApplyMeshConfigDefaults(configYaml)
```

Comment:
The ApplyMeshConfigDefaults function may just override the one you returned from the cluster with default value, i.e. trustDomain=cluster.local here?

Reply:
Default mesh config is applied on fields which is not specified.